ext/standard/tests/general_functions/is_list.phpt

```diff
@@ -0,0 +1,94 @@
+--TEST--
+Test is_list() function
+--FILE--
+<?php
+
+function test_is_list(string $desc, $val) : void {
+	printf("%s: %s\n", $desc, is_list($val) ? "true" : "false");
+}
+
+test_is_list("empty", []);
+test_is_list("one", [1]);
+test_is_list("two", [1,2]);
+test_is_list("three", [1,2,3]);
+test_is_list("four", [1,2,3,4]);
+test_is_list("ten", range(0, 10));
+
+test_is_list("null", null);
+test_is_list("int", 123);
+test_is_list("float", 1.23);
+test_is_list("string", "string");
+test_is_list("object", new stdclass);
+test_is_list("true", true);
+test_is_list("false", false);
+
+test_is_list("string key", ["a" => 1]);
+test_is_list("mixed keys", [0 => 0, "a" => 1]);
+test_is_list("ordered keys", [0 => 0, 1 => 1]);
+test_is_list("shuffled keys", [1 => 0, 0 => 1]);
+test_is_list("skipped keys", [0 => 0, 2 => 2]);
```

Comment:
I would also add a test to cover arrays with negative keys, e.g.: `$arr = [-1 => 1, 0 => 2];`